messageBus/amazonSQS.js

```diff
@@ -0,0 +1,55 @@
+const AWS = require('aws-sdk');
+const config = require('./aws.config');
+
+AWS.config.update({
+  accessKeyId: process.env.SQS_ACCESS_KEY || config.accessKey,
+  secretAccessKey: process.env.SQS_SECRET_KEY || config.secretKey,
+});
+
+const sqs = new AWS.SQS({ region: process.env.SQS_REGION || config.region });
+
+const MAX_NUMBER_OF_MESSAGES_TO_RECEIVE = 10;
+const MESSAGE_VISIBILITY_TIMEOUT = 10;
+const subscribers = [
+  config.queue,
+];
+
+module.exports.publish = (message) => {
+  subscribers.forEach((subscriber) => {
+    const sqsParams = {
+      QueueUrl: subscriber,
+      MessageBody: JSON.stringify(message),
+    };
+    sqs.sendMessage(sqsParams, (err) => {
+      if (err) throw err;
+    });
+  });
+};
+
+module.exports.poll = () => {
+  const sqsParams = {
+    QueueUrl: config.queue,
+    AttributeNames: ['All'],
```

Comment:
Attribute Names is actually optional if you want all of them. I think you are also using the default visibility time out as well.

Reply:
OK, that's good to know -- didn't know about "All" being the default. I'll leave both these in for the time being in case I want to tweak them during development.